v2/azure-arm.yaml

```diff
@@ -1561,6 +1561,51 @@ objectModelConfiguration:
         $exportAs: VirtualNetworksVirtualNetworkPeering
         $supportedFrom: v2.0.0-alpha.1
     2022-07-01:
+      ApplicationGateway:
+        $export: true
+        $supportedFrom: v2.3.0
+      ApplicationGatewayBackendAddressPool:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayBackendHttpSettings:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayBackendSettings:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayFrontendIPConfiguration:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayFrontendPort:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayIPConfiguration:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayHttpListener:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayListener:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayLoadDistributionPolicy:
+         $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayPrivateLinkConfiguration:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayProbe:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayRedirectConfiguration:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayRequestRoutingRule:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayRewriteRuleSet:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayRoutingRule:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewaySslCertificate:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewaySslProfile:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayTrustedClientCertificate:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayTrustedRootCertificate:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayUrlPathMap:
+        $resourceEmbeddedInParent: ApplicationGateway
+      ApplicationGatewayAuthenticationCertificate:
+        $resourceEmbeddedInParent: ApplicationGateway
```

Comment:
Can any of these be created independently, and then referenced by the parent? 

If so, consider using `$isResource` instead to make these available as independent resources. Amongst other factors, this would make the users yaml file easier to read.

Reply:
As shared, the embedded resources are tightly coupled with Appgateway, so my suggestion is to maintain the current implementation.